contracts/btc-light-client/src/msg/contract.rs

```diff
@@ -13,20 +13,37 @@ pub struct InstantiateMsg {
     pub network: babylon_bitcoin::chain_params::Network,
     pub btc_confirmation_depth: u32,
     pub checkpoint_finalization_timeout: u32,
+    /// A sequence of initial BTC headers used to bootstrap the light client.
+    ///
+    /// Must include at least `btc_confirmation_depth` headers following the base header.
+    pub headers: Vec<BtcHeader>,
```

Comment:
Do you think we should make it

```suggestion
    /// Must include at least `btc_confirmation_depth` headers following the base header.
    pub headers: Vec<String>,
```

such that one can instantiate BTC light client contract through CLI?

Reply:
Btw, an alternative here is to take a single `BtcHeaderInfo` for instantiation. The header is picked carefully by the BSN so we don't need to have `k` deep ones. Also if the BSN picks a bad one Babylon won't be synchronising BTC headers with this BSN.

Reply:
I thought about this. So do we still need to instantiate the contract from CLI? I thought we don't need the instantiation from CLI, hence I chose the simplicity here. Do you have any scenarios that we may instantiate the LC contract directly from CLI instead of going through the babylon contract?

>Also if the BSN picks a bad one Babylon won't be synchronising BTC headers with this BSN.

I'm not familar with this part. More input is appreciated.

Reply:
> Do you have any scenarios that we may instantiate the LC contract directly from CLI instead of going through the babylon contract?

Not in our context, but at some point people might take this BTC light client contract for their own use cases. The contract is rather self-contained so I thought we can make it generic

> Also if the BSN picks a bad one Babylon won't be synchronising BTC headers with this BSN.
> I'm not familar with this part. More input is appreciated.

So basically after this PR, Babylon side will also be updated such that upon establishing IBC channel, the contract can send its initial BTC header to Babylon, and Babylon will start forwarding BTC headers since that header. If the BSN chooses an initial header that is invalid, Babylon won't be able to determine the list of BTC headers for forwarding. It's BSN's responsibility to pick a BTC header that is valid and is recent.

That's why we might be able to have a single initial header for simplicity

Reply:
Switched to the single initial header approach, please have another look. A bunch of comments are outdated now and I have resolved them.